generics/middleware/authenticator.js

```diff
@@ -397,127 +391,192 @@ module.exports = async function (req, res, next) {
   };
   // performing default token data extraction
   if (defaultTokenExtraction) {
+    if (!decodedToken.data.organization_id || !decodedToken.data.tenant_id) {
+      rspObj.errCode = reqMsg.TENANT_ORG_MISSING.MISSING_CODE;
+      rspObj.errMsg = reqMsg.TENANT_ORG_MISSING.MISSING_MESSAGE;
+      rspObj.responseCode = responseCode.unauthorized.status;
+      return res.status(responseCode.unauthorized.status).send(respUtil(rspObj));
+    }
+
     userInformation = {
       userToken: token,
       userId: typeof decodedToken.data.id == 'string' ? decodedToken.data.id : JSON.stringify(decodedToken.data.id),
       userName: decodedToken.data.name,
       firstName: decodedToken.data.name,
-      organizationId: decodedToken.data.organization_id,
+      organizationId: decodedToken.data.organization_id.toString(),
+      tenantId: decodedToken.data.tenant_id && decodedToken.data.tenant_id.toString(),
+      roles: decodedToken.data.roles,
     };
   } else {
     // Iterate through each key in the config object
+    let stringTypeKeys = ['userId', 'tenantId', 'organizationId'];
     for (let key in configData) {
       if (configData.hasOwnProperty(key)) {
         let keyValue = getNestedValue(decodedToken, configData[key]);
-        if (key === 'userId') {
+
+        if (stringTypeKeys.includes(key)) {
           keyValue = keyValue.toString();
         }
         // For each key in config, assign the corresponding value from decodedToken
         userInformation[key] = keyValue;
       }
     }
   }
+
+  /**
+   * Validate if provided orgId(s) belong to the tenant by checking against related_orgs.
+   *
+   * @param {String} tenantId - ID of the tenant
+   * @param {String} orgId - Comma separated string of org IDs or 'ALL'
+   * @returns {Object} - Success with validOrgIds array or failure with error object
+   */
+  async function validateIfOrgsBelongsToTenant(tenantId, orgId) {
```

Comment:
@borkarsaish65 As per thediscussion you had with @aks30 all these logic has to be moved to another file right? But I am not seeing that change here